demo/mobile/src/app/initialisation/config.js

```diff
@@ -17,12 +16,29 @@ function createMainAppRoute($stateProvider, $urlRouterProvider, $httpProvider) {
     });
 }
 
-angular.module('wfm-mobile').config(['$stateProvider', '$urlRouterProvider', '$httpProvider', createMainAppRoute]).controller('mainController', [
-  '$rootScope', '$scope', '$state', '$mdSidenav', 'userService', '$window', '$http',
-  function ($rootScope, $scope, $state, $mdSidenav, userService, $window, $http) {
-    userService.getProfile($http, $window).then(function (profileData) {
-      $scope.profileData = profileData;
-    });
+angular.module('wfm-mobile').config(['$stateProvider', '$urlRouterProvider', createMainAppRoute]).controller('mainController', [
+  '$rootScope', '$scope', '$state', '$mdSidenav', '$http', '$window', 'Auth',
+  function($rootScope, $scope, $state, $mdSidenav, $http, $window, Auth) {
+
+    // return user profile from keycloak
+    if (Auth.keycloak) {
+      // display manage account menu item in UI
+      $scope.useKeycloak = true;
+
+      // retrieve the users profile from keycloak
+      var profile = Auth.keycloak.userProfile;
+      $scope.profileData = {
```

Comment:
Antipattern. Can we change original profile method to return actual profile object?
Whathever was called. Code above seems to be little bit strange and it will be hard to wrap that.